src/App.vue

```diff
@@ -464,11 +464,17 @@ export default {
       const tasks = response?.data?.tasks
       if (tasks) {
         this.setTasks(tasks)
+        // special cases for Name Request or Incorp App
         if (this.nrNumber && tasks.length > 0) {
-          if (tasks[0].task?.todo) {
-            this.setEntityStatus(EntityStatus.NAME_REQUEST)
-          } else {
-            this.setEntityStatus(EntityStatus.INCORPORATION_APPLICATION)
+          switch (tasks[0].task?.todo?.header?.name) {
+            case FilingTypes.NAME_REQUEST:
+              // this is a Name Request
+              this.setEntityStatus(EntityStatus.NAME_REQUEST)
+              break
+            case FilingTypes.INCORPORATION_APPLICATION:
+              // this is a Draft Incorporation Application
+              this.setEntityStatus(EntityStatus.DRAFT_INCORP_APP)
```

Comment:
Just curious, what is the difference between a Name Request and Incorporation application. I thought a name request was a draft incorporation application?

Reply:
At the moment, a Name Request does not have a draft (hence "Incorporate Now" vs "Resume"). However, this will shortly change when using a NR creates a draft that we will fetch using Temp Reg Number... this code will need a bit of refactoring and there's a ticket for that.